Sources/NIOExtras/PCAPRingBuffer.swift

```diff
@@ -0,0 +1,98 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+// MARK: NIOPCAPRingBuffer
+/// Storage for the most recent set of packets captured subject to constraints.
+/// Use `addFragment` as the sink to a `NIOWritePCAPHandler` and call `emitPCAP`
+/// when you wish to get the recorded data.
+/// - Warning:  This class is not thread safe so should only be called from one thread.
+public class NIOPCAPRingBuffer {
+    private var pcapFragments: CircularBuffer<ByteBuffer>
+    private var pcapCurrentBytes: size_t
```

Comment:
Any reason we can't use `Int` here and below?

Reply:
Sure - much as I hate Int as a type it does match what we're getting from ByteBuffer.